src/main/java/com/google/api/codegen/transformer/RetryDefinitionsTransformer.java

```diff
@@ -77,23 +101,141 @@
                   interfaceConfigProto.getName()));
         }
       }
-      builder.put(retryDef.getName(), (new ImmutableSet.Builder<String>()).addAll(codes).build());
+      builder.put(retryDef.getName(), (new ImmutableList.Builder<String>()).addAll(codes).build());
+    }
+    if (diagCollector.getErrorCount() > 0) {
+      return null;
+    }
+    return builder.build();
+  }
+
+  /**
+   * Returns a mapping of a retryCodeDef name to the list of retry codes it contains. Also populates
+   * the @param methodNameToRetryCodeNames with a mapping of a Method name to its retry code
+   * settings nmame.
+   */
+  private static ImmutableMap<String, List<String>> createRetryCodesDefinition(
+      DiagCollector diagCollector,
+      Interface apiInterface,
+      ImmutableMap.Builder<String, String> methodNameToRetryCodeNames,
+      ProtoParser protoParser,
+      SymbolTable symbolTable) {
+
+    ImmutableMap.Builder<String, List<String>> builder = ImmutableMap.builder();
+
+    // Check proto annotations for retry settings.
+    for (Method method : apiInterface.getMethods()) {
+
+      Retry retry = protoParser.getRetry(method);
+
+      Set<String> retryCodes = new TreeSet<>();
+      if (protoParser.isHttpGetMethod(method)) {
+        // If this is analogous to HTTP GET, then automatically retry on `INTERNAL` and
+        // `UNAVAILABLE`.
+        retryCodes.addAll(RETRY_CODES_FOR_HTTP_GET);
+      }
+      // Add all retry codes defined in the Retry proto annotation.
+      retryCodes.addAll(retry.getCodesList().stream().map(Code::name).collect(Collectors.toList()));
+
+      // Create a retryCode config internally.
+      String retryCodesName = symbolTable.getNewSymbol(getRetryCodesName(method));
+      builder.put(retryCodesName, (new ImmutableList.Builder<String>()).addAll(retryCodes).build());
+      methodNameToRetryCodeNames.put(method.getSimpleName(), retryCodesName);
     }
+
     if (diagCollector.getErrorCount() > 0) {
       return null;
     }
     return builder.build();
   }
 
+  /**
+   * Returns a mapping of a retryCodeDef name to the list of retry codes it contains. Also populates
+   * the @param methodNameToRetryCodeNames with a mapping of a Method name to its retry code
+   * settings nmame.
+   */
+  public static ImmutableMap<String, List<String>> createRetryCodesDefinition(
```

Comment:
This is a lot of new code that seems to be doing similar things. Can we comment on the differences? Are some of these temporary until we can move off gapic.yaml? etc.